### PR TITLE
PostCSS support

### PR DESCRIPTION
**Pull request by chuckdumont**
Use PostCSS to run the minifier and configured plugins.  Supports running the same plugins both in the aggregator and in the css loader plugin when not using the aggregator. 

Plugins are configured in the AMD loader config on the client and/or the server-side AMD config on the server using the `postcss.plugins` property specifying an array of two element arrays as in the following example:

```
postcss: {
	plugins: [
		[
			'autoprefixer',  // Module id of the plugin resource
			function(autoprefixer) { 
				return autoprefixer({browsers: '> 1%'}).postcss; // the init function
			}
		]
	]
},
```

### CODE REVIEW

**Comment by KenPowers** [comment]
Are you relying on, in this example, `autoprefixer` being available in the global scope? Taking a look at the browserify standalone loader:

```js
    if ("object" == typeof exports && "undefined" != typeof module) module.exports = e();
    else if ("function" == typeof define && define.amd) define([], e);
    else {
        var f;
        "undefined" != typeof window ? f = window : "undefined" != typeof global ? f = global : "undefined" != typeof self && (f = self), f.postcss = e()
    }
```

Each case is mutually exclusive. Maybe we want to run an AMD loader in Rhino in order to normalize across Rhino and Browsers?

**Comment by chuckdumont** [comment]
@KenPowers,Not sure what the concern is you're raising.  The 'global' variable in global scope is set to refer to the global scope, in the same way that the window variable refers to the global scope in browsers.  That's done with this code:

```
	// Set "global" variable to point to shared scope
	sharedScope.put("global", sharedScope, sharedScope); //$NON-NLS-1$
```

**Comment by KenPowers** [comment]
What I'm saying is that that'll work in Rhino, but in the browser there won't be a global object and the browserify code will use the page's AMD loader but the initialize function won't have a reference.

**Comment by chuckdumont** [comment]
Well, in the browser, won't the code just use window instead of global, with the same results?

**Comment by KenPowers** [comment]
No, take a look at the browserify code again. The cases of loading for Common JS (`global` object), AMD, and browsers (`window` object) are mutually exclusive. It will see the AMD loader in scope and use that and then never reach the code to attach to `window`.

**Comment by chuckdumont** [comment]
OK, I get it.  I'd really rather not load require in rhino if we don't need to.  We could just pass in the module variable to the initialization function.  On the browser, where we used require to load the module, it would have a value, but in JAGGR, it wouldn't.  So then the config code would look like this:

```
postcssPlugins: [
    [
        'autoprefixer',  // Module id of the plugin resource
        function(m) { return (m||autoprefixer)({browsers: '> 1%'}).postcss; } // the init function
    ]
],
```

**Comment by KenPowers** [comment]
That seems like a really hacky work-around to me. It creates extra work for users of JAGGR. What if you just made a shim AMD loader in JAGGR since we shouldn't need to manage dependencies anyway:

```js
var module;
function define(f) {
  module = f();
}
```

And then you could call the initialization function with `module`.

**Comment by chuckdumont** [comment]
That makes me really nervous because it could lead modules to believe that an AMD loader is available when it isn't and they may try to do other things with AMD that would break.

**Comment by KenPowers** [comment]
Then I'd rather use a real AMD loader. If you ask me requiring the client to account for different loaders isn't the way to go.

**Comment by chuckdumont** [comment]
OK, so while were at it, we might as well support node.js style module.exports.  The following variation on your define shim would work for both node.js and AMD.

```
var module = {exports:0}; 
var define = function(deps, f){
    if (deps && deps.length) throw new Error('define dependencies not supported.');
    module.exports=f();
};
define.amd = true;
```

**Comment by chuckdumont** [comment]
Changes pushed, please review.

File: jaggr-core/src/main/java/com/ibm/jaggr/core/impl/config/ConfigImpl.java [hunk 1]
```diff
@@ -127,8 +127,8 @@ public ConfigImpl(IAggregator aggregator) throws IOException {
 			callConfigModifiers(rawConfig);
 
 			// Seal the config object and the shared scope to prevent changes
-			((ScriptableObject)rawConfig).sealObject();
-			((ScriptableObject)sharedScope).sealObject();
+			sealObject((ScriptableObject)rawConfig);
+			sealObject((ScriptableObject)sharedScope);
```
Comment (KenPowers) [inline jaggr-core/src/main/java/com/ibm/jaggr/core/impl/config/ConfigImpl.java]: image

File: jaggr-sample/src/test/javascript/testCss.js [hunk 2]
```diff
@@ -120,4 +120,42 @@ define([ 'dojo/_base/window', 'js/css'], function(win, css) {
 				.toBe("@import 'foo/bar/themes/images/smile.png?arg1=val1&arg2=val2';");
 		});
 	});
+	
+	describe("PortCSS extension tests", function() {
```
Comment (KenPowers) [inline jaggr-sample/src/test/javascript/testCss.js]: PostCSS?

**Pull request merged**